webapp/components/PollutionMapBase.tsx

```diff
@@ -52,6 +66,47 @@ export default function PollutionMapBaseLayer({
     }
   }
 
+  useEffect(() => {
+    if (mapRef && centerOnZone !== ZONE_NOZONE) {
+      setZone(centerOnZone);
+
+      let newMapCenter: number[];
+      let newZoomFactor: number;
+
+      switch (centerOnZone) {
+        case ZONE_GUADELOUPE:
+          newMapCenter = [-61.5, 16.2];
+          newZoomFactor = 9;
+          break;
+        case ZONE_LAREUNION:
+          newMapCenter = [55.5, -21.2];
+          newZoomFactor = 8;
+          break;
+        case ZONE_MARTINIQUE:
+          newMapCenter = [-61, 14.7];
+          newZoomFactor = 9;
+          break;
+        case ZONE_MAYOTTE:
+          newMapCenter = [45, -12.75];
+          newZoomFactor = 9;
+          break;
+        case ZONE_GUYANE:
+          newMapCenter = [-52.7, 4.3];
+          newZoomFactor = 6;
+          break;
+        case ZONE_METROPOLE:
+        default:
+          newMapCenter = [2.5, 46.2];
+          newZoomFactor = 5;
+          break;
+      }
+
+      if (mapRef.current && zone !== ZONE_NOZONE) {
```

Comment:
:mag: Cette condition étant déjà vérifié plus haut, n'est ptt pas nécessaire ici
```suggestion
      if (mapRef.current) {
```

Reply:
Exact.